src/room/useLoadGroupCall.ts

```diff
@@ -38,51 +45,196 @@ export type GroupCallLoading = {
   kind: "loading";
 };
 
+export type GroupCallWaitForInvite = {
+  kind: "waitForInvite";
+  roomSummary: RoomSummary;
+};
+
+export type GroupCallCanKnock = {
+  kind: "canKnock";
+  roomSummary: RoomSummary;
+  knock: () => void;
+};
+
 export type GroupCallStatus =
   | GroupCallLoaded
   | GroupCallLoadFailed
-  | GroupCallLoading;
+  | GroupCallLoading
+  | GroupCallWaitForInvite
+  | GroupCallCanKnock;
 
-export interface GroupCallLoadState {
-  error?: Error;
-  groupCall?: GroupCall;
+export class CustomMessage extends Error {
+  public messageBody: string;
+  public reason?: string;
+  public constructor(
+    messageTitle: string,
+    messageBody: string,
```

Comment:
I had to look further down into the code to tell whether these fields store translated strings, so that would be good to comment about